tests/getById/endpoint.test.js

```diff
@@ -6,35 +6,150 @@ var config = require('../../config/load-config.js');
   initServer(config, function (error, server, pool) {
     if (error) {
       return t.fail('Error starting the server, error: ', error);
     }
 
-    return server.inject({ method: 'GET', url: '/getById' },
-      function (res) {
-        var expected = {
-          creator_id: 3,
-          description: 'How can I...?',
-          id: 2,
-          org_id: 1,
-          tags: [
-            {
-              id: 2,
-              name: 'Corporate'
-            }
-          ],
-          title: 'Challenge Number 2'
-        };
-
-
-        t.deepEquals(res.result, expected,
-          'getById returns challenge details and associated tag ids');
-
-        return pool.end(function () {
-          server.stop(t.end);
-        });
+    return server.inject(endpoint(2), function (res) {
+      var expected = {
+        creator_id: 3,
+        description: 'How can I...?',
+        id: 2,
+        org_id: 1,
+        org_name: 'Apple',
+         tags: [
+          {
+            id: 2,
+            name: 'Corporate'
+          }
+        ],
+        title: 'Challenge Number 2'
+      };
+
+      t.deepEquals(res.result, expected,
+        'returns challenge details and associated tag ids');
+
+      return pool.end(function () {
+        server.stop(t.end);
+      });
+    });
+  });
+});
+
+// org.active = true and chal.active = true. No tags
+test('getById for challengeId = 5', function (t) {
+  initServer(config, function (error, server, pool) {
+    if (error) {
+      return t.fail('Error starting the server, error: ', error);
+    }
+
+    return server.inject(endpoint(5), function (res) {
+      var expected = {
+        creator_id: 4,
+        description: 'How have...?',
+        id: 5,
+        org_id: 2,
+        org_name: 'dwyl',
+        tags: [],
+        title: 'Challenge Number 5'
+      };
+
+      t.deepEquals(res.result, expected,
+        'returns challenge details and empty array for tags');
+
+      return pool.end(function () {
+        server.stop(t.end);
+      });
+    });
+  });
+});
+
+// org.active = true, chal.active = false. No tags
+test('getById for challengeId = 9', function (t) {
+  initServer(config, function (error, server, pool) {
+    if (error) {
+      return t.fail('Error starting the server, error: ', error);
+    }
+
+    return server.inject(endpoint(9), function (res) {
+      var expected = '[]';
+
+      t.deepEquals(res.payload, expected,
+        'empty object returned for inactive chal of an active org');
+
+      return pool.end(function () {
+        server.stop(t.end);
+      });
+    });
+  });
+});
+
+// org.active = false, chal.active = false. No tags
+test('getById for challengeId = 10', function (t) {
+  initServer(config, function (error, server, pool) {
+    if (error) {
+      return t.fail('Error starting the server, error: ', error);
+    }
+
+    return server.inject(endpoint(10), function (res) {
+      var expected = '[]';
+
+      t.deepEquals(res.payload, expected,
```

Comment:
I've had to change the test from 

`t.deepEquals(res.result, expected)` where `var expected = []` 

to  

`t.deepEquals(res.payload, expected)` where `var expected = '[]'`

because if you `t.deepEquals({}, [])` it comes back as true! :boom: